public/app/features/manage-dashboards/components/DashboardImport.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { FormEvent, PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { hot } from 'react-hot-loader';
+import { css } from 'emotion';
+import { AppEvents, NavModel } from '@grafana/data';
+import { Forms, stylesFactory } from '@grafana/ui';
+import Page from 'app/core/components/Page/Page';
+import ImportDashboardForm from './ImportDashboardForm';
+import { DashboardFileUpload } from './DashboardFileUpload';
+import { fetchGcomDashboard, importDashboardJson } from '../state/actions';
+import appEvents from 'app/core/app_events';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { StoreState } from 'app/types';
+
+interface Props {
+  navModel: NavModel;
+  isLoaded: boolean;
+  fetchGcomDashboard: typeof fetchGcomDashboard;
+  importDashboardJson: typeof importDashboardJson;
+}
+
+interface State {
+  gcomDashboard: string;
+  dashboardJson: string;
+  dashboardFile: any;
+  gcomError: string;
+}
+
+const importStyles = stylesFactory(() => {
+  return {
+    option: css`
+      margin-bottom: 32px;
+    `,
+  };
+});
+
+class DashboardImport extends PureComponent<Props, State> {
```

Comment:
Would it be a good idea to use the `unconnected`-convention together with the `mapStateToProps` function that Torkel talked about on frontend bi-weekly?

I think we have an example on how it is used here: https://github.com/grafana/grafana/blob/e22798dea8fb6285a3c7a0995e287c3098981e74/public/app/features/dashboard/components/SubMenu/SubMenu.tsx

Reply:
Ah, yes